src/main/java/org/reaktivity/command/log/internal/LoggableStream.java

```diff
@@ -212,32 +245,35 @@ private void onBegin(
         final String targetName = labels.lookupLabel(targetId);
 
         out.printf(streamFormat, index, offset, timestamp, traceId, sourceName, targetName, routeId, streamId,
-                   format("BEGIN [0x%016x] [0x%016x]", authorization, affinity));
+            format("BEGIN [0x%016x] [0x%016x]", authorization, affinity));
 
-        if (verbose)
+
+        final ExtensionFW extension = begin.extension().get(extensionRO::tryWrap);
+        if (extension != null)
         {
-            final ExtensionFW extension = begin.extension().get(extensionRO::tryWrap);
-            if (extension != null)
+            final Consumer<BeginFW> beginHandler = beginHandlers.get(extension.typeId());
+            if (beginHandler != null)
             {
-                final Consumer<BeginFW> beginHandler = beginHandlers.get(extension.typeId());
-                if (beginHandler != null)
-                {
-                    beginHandler.accept(begin);
-                }
+                beginHandler.accept(begin);
             }
         }
     }
 
     private void onData(
-        final DataFW data)
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
     {
+        final DataFW data = dataRO.wrap(buffer, index, index + length);
+
         final int offset = data.offset() - HEADER_LENGTH;
         final long timestamp = data.timestamp();
         final long routeId = data.routeId();
         final long streamId = data.streamId();
         final long traceId = data.traceId();
         final long budgetId = data.budgetId();
-        final int length = data.length();
+        final int dataLength = data.length();
```

Comment:
After applying feedback to make this method strongly typed, please revert this rename as it is no longer required.